lib/orbital/api.rb

```diff
@@ -142,6 +144,82 @@ def search_payment_methods(search_key, offset, limit, properties, context)
+        plugin_trx_info.status == :UNDEFINED && pass_delay_time(plugin_trx_info, options)
+      end
+
+      def fix_undefined_trx(plugin_trx_info, context, options)
+        inquiry_response = inquiry plugin_trx_info, context
+        update_response_if_needed plugin_trx_info, inquiry_response, options
+      end
+
+      def update_response_if_needed(plugin_trx_info, inquiry_response, options)
+        response_id = find_value_from_properties(plugin_trx_info.properties, 'orbital_response_id')
+        response = OrbitalResponse.find_by(:id => response_id)
+        updated = false
+        if should_update_response inquiry_response, plugin_trx_info
+          logger.info("Fixing UNDEFINED kb_transaction_id='#{plugin_trx_info.kb_transaction_payment_id}', success='#{inquiry_response.success?}'")
+          response.update_and_create_transaction(inquiry_response)
+          updated = true
+        elsif should_cancel_payment plugin_trx_info, options
+          @logger.info("Canceling UNDEFINED kb_transaction_id='#{plugin_trx_info.kb_transaction_payment_id}'")
+          response.cancel
+          updated = true
+        end
+        updated
+      end
+
+      def should_update_response(inquiry_response, plugin_trx_info)
+        !inquiry_response.nil? && !inquiry_response.params.nil? && inquiry_response.params['order_id'] == plugin_trx_info.second_payment_reference_id
+      end
+
+      def should_cancel_payment(plugin_trx_info, options)
+        threshold = (Killbill::Plugin::ActiveMerchant::Utils.normalized(options, :cancel_threshold) || ONE_HOUR_AGO).to_i
+        delay_since_trx(plugin_trx_info) >= threshold
+      end
+
+      def pass_delay_time(plugin_trx_info, options)
+        janitor_delay_threshold = (Killbill::Plugin::ActiveMerchant::Utils.normalized(options, :janitor_delay_threshold) || FIVE_MINUTES_AGO).to_i
+        delay_since_trx(plugin_trx_info) >= janitor_delay_threshold
+      end
+
+      def delay_since_trx(plugin_trx_info)
+        delay_since_transaction = now - plugin_trx_info.created_date
+        delay_since_transaction < 0 ? 0 : delay_since_transaction
+      end
+
+      def now
+        Time.parse(@clock.get_clock.get_utc_now.to_s)
+      end
+
+      def inquiry(plugin_trx_info, context)
+        payment_processor_account_id = find_value_from_properties(plugin_trx_info.properties, 'payment_processor_account_id')
+        trace_number = find_value_from_properties(plugin_trx_info.properties, 'trace_number')
+        orbital_order_id = plugin_trx_info.second_payment_reference_id
+
+        return nil if trace_number.nil? || orbital_order_id.nil? || payment_processor_account_id.nil?
+
+        gateway = lookup_gateway(payment_processor_account_id, context.tenant_id)
+        gateway.inquiry(orbital_order_id, trace_number)
+      end
+
+      def with_trace_num_and_order_id(properties)
+        {:params_trace_number => find_value_from_properties(properties, :trace_number),
```

Comment:
It looks like Orbital requires two `ids` to retrieve payment information? What is the difference between the two?

Reply:
One Order Id can have multiple trace numbers that correspond to each request we send.